mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -38,66 +38,69 @@
  *
  */
 
-#include "mbf_abstract_nav/abstract_controller_execution.h"
 #include <mbf_msgs/ExePathResult.h>
 #include <mbf_utility/thread_affinity.h>
 
+#include "mbf_abstract_nav/abstract_controller_execution.h"
+
 namespace mbf_abstract_nav
 {
 
-  const double AbstractControllerExecution::DEFAULT_CONTROLLER_FREQUENCY = 100.0; // 100 Hz
-
-  AbstractControllerExecution::AbstractControllerExecution(
-      const std::string name,
-      const mbf_abstract_core::AbstractController::Ptr& controller_ptr,
-      const ros::Publisher& vel_pub,
-      const ros::Publisher& goal_pub,
-      const TFPtr &tf_listener_ptr,
-      const MoveBaseFlexConfig &config) :
-    AbstractExecutionBase(name),
-      controller_(controller_ptr), tf_listener_ptr(tf_listener_ptr), state_(INITIALIZED),
-      moving_(false), max_retries_(0), patience_(0), vel_pub_(vel_pub), current_goal_pub_(goal_pub),
-      calling_duration_(boost::chrono::microseconds(static_cast<int>(1e6 / DEFAULT_CONTROLLER_FREQUENCY)))
-  {
-    ros::NodeHandle nh;
-    ros::NodeHandle private_nh("~");
-
-    // non-dynamically reconfigurable parameters
-    private_nh.param("robot_frame", robot_frame_, std::string("base_link"));
-    private_nh.param("map_frame", global_frame_, std::string("map"));
-    private_nh.param("mbf_tolerance_check", mbf_tolerance_check_, false);
-    private_nh.param("dist_tolerance", dist_tolerance_, 0.1);
-    private_nh.param("angle_tolerance", angle_tolerance_, M_PI / 18.0);
-    private_nh.param("tf_timeout", tf_timeout_, 1.0);
-
-    // dynamically reconfigurable parameters
-    reconfigure(config);
-  }
-
-  AbstractControllerExecution::~AbstractControllerExecution()
-  {
-  }
+const double AbstractControllerExecution::DEFAULT_CONTROLLER_FREQUENCY = 100.0; // 100 Hz
+
+AbstractControllerExecution::AbstractControllerExecution(
+    const std::string &name,
+    const mbf_abstract_core::AbstractController::Ptr &controller_ptr,
+    const ros::Publisher &vel_pub,
+    const ros::Publisher &goal_pub,
+    const TFPtr &tf_listener_ptr,
+    const MoveBaseFlexConfig &config) :
+  AbstractExecutionBase(name),
+    controller_(controller_ptr), tf_listener_ptr(tf_listener_ptr), state_(INITIALIZED),
+    moving_(false), max_retries_(0), patience_(0), vel_pub_(vel_pub), current_goal_pub_(goal_pub),
+    calling_duration_(boost::chrono::microseconds(static_cast<int>(1e6 / DEFAULT_CONTROLLER_FREQUENCY)))
+{
+  ros::NodeHandle nh;
+  ros::NodeHandle private_nh("~");
+
+  // non-dynamically reconfigurable parameters
+  private_nh.param("robot_frame", robot_frame_, std::string("base_link"));
+  private_nh.param("map_frame", global_frame_, std::string("map"));
+  private_nh.param("force_stop_at_goal", force_stop_at_goal_, false);
+  private_nh.param("force_stop_on_cancel", force_stop_on_cancel_, false);
+  private_nh.param("mbf_tolerance_check", mbf_tolerance_check_, false);
```

Comment:
these two params `force_stop_at_goal` and `force_stop_on_cancel` is something we implemented as well. I think we need to turn them on explicitly either set them to `true` here or we can do it in roslaunch file when we launch move base flex so that we can configure it. We should also QA test that it stops at goal immediately by monitoring velocity.

Reply:
how do you mean we implemented them as well? is there a duplicate parameter?